### ISSUE TITLE
Build failing on umbrella projects

### ISSUE THREAD

**Issue by costaraphael** [JakeBecker/elixir-ls#6]
Here is the logged output:

```
[Info  - 12:38:36] Child Logger.ErrorHandler of Supervisor Logger.Supervisor started
Pid: #PID<0.78.0>
Start Call: Logger.Watcher.watcher(:error_logger, Logger.ErrorHandler, {true, true, 500})
Restart: :permanent
Shutdown: 5000
Type: :worker
[Info  - 12:38:36] Application logger started at :nonode@nohost
[Info  - 12:38:37] Child ElixirLS.LanguageServer.Builder of Supervisor ElixirLanguageServer.Supervisor started
Pid: #PID<0.83.0>
Start Call: ElixirLS.LanguageServer.Builder.start_link(ElixirLS.LanguageServer.Builder)
Restart: :permanent
Shutdown: 5000
Type: :worker
[Info  - 12:38:37] Child ElixirLS.LanguageServer.Server of Supervisor ElixirLanguageServer.Supervisor started
Pid: #PID<0.84.0>
Start Call: ElixirLS.LanguageServer.Server.start_link(ElixirLS.LanguageServer.Server)
Restart: :permanent
Shutdown: 5000
Type: :worker
[Info  - 12:38:37] Child ElixirLS.IOHandler of Supervisor ElixirLanguageServer.Supervisor started
Pid: #PID<0.85.0>
Start Call: ElixirLS.IOHandler.start_link(ElixirLS.LanguageServer.Server, [name: ElixirLS.LanguageServer.IOHandler])
Restart: :permanent
Shutdown: 5000
Type: :worker
[Info  - 12:38:37] Application language_server started at :nonode@nohost
[Info  - 12:38:37] Forcing full rebuild
warning: redefining module EF.Mixfile (current version defined in memory)
  mix.exs:1

[Info  - 12:38:54] Forcing full rebuild
warning: redefining module EF.Mixfile (current version defined in memory)
  mix.exs:1

[Info  - 12:38:55] Forcing full rebuild
warning: redefining module EF.Mixfile (current version defined in memory)
  mix.exs:1

[Info  - 12:39:02] Forcing full rebuild
[Warn  - 12:39:02] Build failed: %RuntimeError{message: "trying to access Mix.Project.app_path for an umbrella project but umbrellas have no app"}
[Info  - 12:39:02] Forcing full rebuild
warning: redefining module EF.Mixfile (current version defined in memory)
  mix.exs:1

[Warn  - 12:39:02] Build failed: %RuntimeError{message: "trying to access Mix.Project.app_path for an umbrella project but umbrellas have no app"}
```

I looked around at the server code and I guess Mix.Project.app_path is being called inside some of the other calls to the Mix.Project module during the project compilation.

I could take a look at this, if you don't mind. And any directions are appreciated. :smile:

**Comment by sjmueller** [comment]
@JakeBecker we need support for umbrella projects too. Do you have any insights on how to get this working?

**Comment by costaraphael** [comment]
I took a look at the Mix's `compile` task and the code has no verification to whether the project is an umbrella or not. I think that's because Mix detects that it is in an umbrella project before the tasks are run, and runs the task on each project.

Is it feasible to do something like that (compile each app individually when inside an umbrella)?

There is a `Mix.Project.umbrella?` function that may help with this.

**Comment by JakeBecker** [comment]
Apologies for the wait on this issue. Thanks for bearing with me.

I've rewritten the build system after contributing some changes to Elixir 1.6. Details here: https://medium.com/@JakeBeckerCode/elixirls-0-2-better-builds-code-formatter-and-incremental-dialyzer-be70999ea3e7

If you update the VS Code plugin, this issue should be fixed, though you'll need Elixir 1.6 installed to get build diagnostics in the editor. Please open a new issue if it's still broken in some way.

Thank you!

**Comment by sjmueller** [comment]
You're a rock star @JakeBecker!! Just read the blog post and this looks amazing, can't wait to try the new stuff out!
